### DISCUSSION TITLE
Passing url parameters to systems like Zitadel

### DISCUSSION THREAD

**Discussion by wtcross** [erlef/oidcc_plug#22] [Q&A] [answered]
Zitadel is an open source identity and access management solution. I noticed [maennchen/zitadel_api](https://github.com/maennchen/zitadel_api) so I'm assuming @maennchen might have implemented what I'm trying to do before.

The Zitadel docs indicate that adding a `prompt=create` URL parameter to the authorization redirect URL will immediately show the registration page in Zitadel. I have been trying to use [Oidcc.Plug.Authorize](https://hexdocs.pm/oidcc_plug/Oidcc.Plug.Authorize.html#t:opts/0) in my oidcc controller (created by [erlef/phx_gen_oidcc](https://github.com/erlef/phx_gen_oidcc)) by passing `url_extension: [{<<"prompt">>, <<"create">>}]` as part of the plug options, but it is not working. The prompt URL parameter is not added to the redirect URL.

Am I missing something obvious?

**Comment by wtcross** [comment]
I forgot to share a link to the Zitadel documentation that I have been referencing. Here it is:
https://zitadel.com/docs/guides/integrate/onboarding/end-users#prompt-registration-form

**Comment by maennchen** [comment]
You should be on the right track. There's probably a bug. I'll have a look into it :+1:

**Reply by maennchen** [reply, marked as answer]
Does this PR solve your issue? https://github.com/erlef/oidcc/pull/378

**Reply by maennchen** [reply]
@wtcross Released as v3.2.4.

**Comment by wtcross** [comment]
Thank you so much for the rapid response. That PR does fix the issue! I am very rusty after having not writing any code for ~2 years and am picking a side project back up. Once I get ramped up again I'll hopefully start contributing to these projects. My Erlang knowledge has faded :sweat_smile: , but it'll come back.

**Comment by wtcross** [comment]
Your oidcc libraries were very easy to get up and running with on a Phoenix application. Thank you for all the work. I remember there being a gap in the elixir ecosystem from an OpenID Connect perspective when I was last working on this project.

**Reply by maennchen** [reply]
Great to hear that you like working with them. :smile:
